lib/retry.js

```diff
@@ -87,14 +102,17 @@ export default function retry(opts, task, callback) {
             acc.intervalFunc = typeof t.interval === 'function' ?
                 t.interval :
                 constant(+t.interval || DEFAULT_INTERVAL);
+
+            if(typeof t.filter === 'function') {
+                acc.filter = t.filter;
+            }
```

Comment:
no need for the conditional just set it here and check if it is a function when you execute